src/test/java/test/assertion/SoftAssertTest.java

```diff
@@ -35,6 +35,34 @@ public void onAssertFailure(IAssert assertCommand, AssertionError ex) {
     Assert.assertEquals(failures.size(), 1, failures.toString());
   }
 
+
+  @Test
+  public void testDefaultMessage() {
+    try {
+      final SoftAssert sa = new SoftAssert();
+      sa.assertTrue(false);
+      sa.assertAll();
+      Assert.fail();
+    } catch (AssertionError exc) {
+      Assert.assertNotNull(exc.getMessage());
+      Assert.assertFalse(exc.getMessage().isEmpty());
+    }
+  }
+
+  @Test
+  public void testCustomMessage() {
+    String msg = "GeneralMessage";
+    try {
+      final SoftAssert sa = new SoftAssert();
+      sa.assertTrue(false);
+      sa.assertAll(msg);
+      Assert.fail();
+    } catch (AssertionError exc) {
+      Assert.assertNotNull(exc.getMessage());
+      Assert.assertTrue(exc.getMessage().startsWith(msg));
+    }
+  }
+
```

Comment:
You could avoid the duplication with one test and one data provider providing two messages, one null and one non null.